src/lib/hash.c

```diff
@@ -271,3 +271,50 @@ pgp_digest_length(pgp_hash_alg_t alg, size_t *output_length)
     botan_hash_destroy(handle);
     return ret;
 }
+
+bool
+pgp_hash_list_add(list *hashes, pgp_hash_alg_t alg)
```

Comment:
comment here would be useful. Something that tells me what I should do if this function returns false. In more details - if ``pgp_hash_create`` fails - then memory is not allocated, but if ``list_append`` fails then memory is allocated and I need to free it with.

Reply:
Thanks for spotting this. Actually, I forgot to `pgp_hash_finish()` in case of `list_append()` fails. Will fix.